react/week-2/lesson.md

```diff
@@ -429,6 +437,57 @@ To help us cleanly split up code that performs business logic from code that sho
 
 Container components usually have some state and handler methods. Because of this they must use the `class` syntax. Presentational components on the other hand don't require the more verbose syntax. Instead they usually use the functional syntax.
 
+## Fetching data in React
+
+Often when you create a React app, you will want to get data from an API, and display it inside your components.
+How do we do this in React? Where does the API call go, and when should we trigger it?
+
+**Where:** Most likely at the top of the component tree, inside a 'container' component as explained above. You can then flow the data down into your child components as props.
+**When:** When the component is first loaded into the DOM. We call this 'mounting'.
+**How:** With a handy new hook called `useEffect`
+
+### The `useEffect` hook
+
+Just like `useState`, the `useEffect` hook is a special function that all function components have access to. This is the syntax to follow to fetch data on mount:
+
+```js
+useEffect(() => {
+  // Make your API call here!
+}, []); // Don't forget these empty square brackets
+```
+
+And here is a more complete example:
+
+```js
+import React, { useState, useEffect } from "react";
+
+const TvShowContainer = () => {
+  const [movies, setMovies] = useState([]);
+
+  useEffect(() => {
+    fetch("http://example.com/movies.json").then((response) => {
+      setMovies(response.json());
+    });
+  });
+
+  return (
+    <div>
+      {movies.length > 0 ? (
+        <div>There are {movies.length} to display</div>
+      ) : (
+        <p>Sorry, there are no movies to display</p>
```

Comment:
What do you think about teaching an explicit loading state?

Although I would probably separate that from the simple data fetching example here.

Reply:
Oh actually, looking through my changes I have actually covered some stuff on loading state. So perhaps not necessary.